File: PythonSolutions/SwapNodesInPairs.py
from typing import Optional


# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None or head.next is None:
            return head
        curr = head
        prev = None
        head2 = None
        while curr is not None and curr.next is not None:
            temp = curr.next
            curr.next = temp.next
            temp.next = curr
            if head2 is None:
                head2 = temp
            elif prev is not None:
                prev.next = temp
            prev = curr
            curr = curr.next
        return head2

File: C#-Solutions/SwapNodesInPairs.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode SwapPairs(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }
        ListNode curr = head;
        ListNode prev = null;
        ListNode head2 = null;

        while (curr != null && curr.next != null) {
            ListNode temp = curr.next;
            curr.next = temp.next;
            temp.next = curr;
            if (head2 == null)
                head2 = temp;
            else if (prev != null)
                prev.next = temp;

            prev = curr;
            curr = curr.next;
        }
        return head2;
    }
}